### ISSUE TITLE
Condense results to eliminate `null` periods

### ISSUE THREAD

**Issue by DaveSkender** [DaveSkender/Stock.Indicators#833]
I'd like a utility to remove all of the `null` and `NaN` periods from the results.

Most `results` return the same number of periods as the quotes provided.  This requires the use of `null` values that pre-prepend the start of most time-series results and use `null` or `NaN` result values when incalculable.

This is utility is essentially the opposite action as the internal `SyncIndex()` model in #832 where `null` periods are added to ensure comparable collections are of the same length with matching indexed dates.

For chainable results, this could be as easy as something like:

```csharp
    // CONDENSE (REMOVE null and NaN results)
    public static IEnumerable<TResult> Condense<TResult>(
        this IEnumerable<TResult> results)
        where TResult : IReusableResult
    {
        List<TResult> resultsList = results
            .ToList();

        _ = resultsList
            .RemoveAll(match:
                x => x.Value is null || x.Value is double and double.NaN);

        return resultsList.ToSortedList();
    }
```

But for results that do not have a primary output value, like `AlligatorResult`, a custom `Condense` will be required to handle the unique scenario where all three outputs would need to be `null` to be identified for removal.

**Comment by DaveSkender** [comment]
For chainable results, this could be as easy as something like:

```csharp
    // CONDENSE (REMOVE null and NaN results)
    public static IEnumerable<TResult> Condense<TResult>(
        this IEnumerable<TResult> results)
        where TResult : IReusableResult
    {
        List<TResult> resultsList = results
            .ToList();

        _ = resultsList
            .RemoveAll(match:
                x => x.Value is null || x.Value is double and double.NaN);

        return resultsList.ToSortedList();
    }
```

But for results that do not have a primary output value, like `AlligatorResult`, a custom `Condense` will be required to handle the unique scenario where all three outputs would need to be `null` to be identified for removal.

**Comment by DaveSkender** [comment]
This was released in v2.0.3
